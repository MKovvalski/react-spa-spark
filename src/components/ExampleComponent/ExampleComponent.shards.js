import styled   from '@emotion/styled'
import { down } from 'styled-breakpoints'

export const PositionWrapper = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px;
  font-size: 20px;
  margin-top: 15px;
`

export const MainTitle = styled.h3`
  font-size: ${({ theme }) => theme.contentSize.header};
  color: ${({ theme }) => theme.colors.primary};
  margin-bottom: 15px;
  transition: color .5s;
  ${down('lg')} {
    color: ${({ theme }) => theme.colors.secondary};
  }
`
export const WhatsNew = styled.ul`
  list-style-type: circle;
  font-size: ${({ theme }) => theme.contentSize.list};
  margin-bottom: 15px;
  color: ${({ toggled, theme }) => toggled ? theme.colors.primary : theme.colors.defaultBlack};
`

export const ListElement = styled.li`
  margin-bottom: 5px;
  font-family: ${({ theme }) => theme.fonts.secondary};
`
