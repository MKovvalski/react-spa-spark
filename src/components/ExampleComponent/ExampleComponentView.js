import React, { useState } from 'react'
import {
  MainTitle,
  PositionWrapper,
  WhatsNew,
  ListElement
}                          from './ExampleComponent.shards'
import { CounterButton }   from '../Commons/Buttons.shards'

const ExampleComponent = () => {

  const [toggled, toggleColor] = useState(false)
  return (
    <PositionWrapper>
      <MainTitle>
        Aloha and welcome to React SPA spark!
      </MainTitle>
      <WhatsNew toggled={toggled}>
        <ListElement>emotion replaced sass</ListElement>
        <ListElement>stylelint replaced sass-lint</ListElement>
        <ListElement>removed code overhead from webpack configs</ListElement>
        <ListElement>created a example file structure for emotion</ListElement>
      </WhatsNew>
      <CounterButton onClick={() => toggleColor(!toggled)}>
        I am a common button
      </CounterButton>
    </PositionWrapper>
  )
}

export default ExampleComponent
