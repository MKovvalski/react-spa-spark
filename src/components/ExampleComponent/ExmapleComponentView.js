import React, { useState } from 'react'

const ExampleComponent = () => {

  const [count, setCount] = useState(0)

  // TODO: add state test
  // add base methods for shallow and so forth
  // test conditional rendering
  return (
    <div className='example-component-class'>
      <h2>I am an Example Component</h2>
      <div>Build to test and showcase capabilities of jest and enzyme</div>
      <div
        className='example-component-class__button'
        onClick={() => setCount(count + 1)}
      >
        Up the Counter!
      </div>
      <div>
        Display number of counter clicks:
        <span>{count}</span>
      </div>
    </div>
  )
}

export default ExampleComponent
