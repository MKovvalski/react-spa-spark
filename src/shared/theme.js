// place all variables (colors, shadows, backgrounds) for your project here
import breakpoints from './breakpoints'

const colors = {
  primary: '#6699ff',
  secondary: '#00cc66',
  defaultBlack: '#000000'
}

const contentSize = {
  header: '34px',
  subHeader: '24px',
  list: '20px'
}

const fonts = {
  base: 'Helvetica Neue',
  secondary: 'sans-serif'
}
const theme = {
  breakpoints,
  colors,
  contentSize,
  fonts
}

export default theme
