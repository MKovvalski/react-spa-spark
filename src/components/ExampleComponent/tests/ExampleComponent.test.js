import React             from 'react'
import { render }        from 'enzyme'
import { ThemeProvider } from 'emotion-theming'
import ExampleComponent  from '../index'
import theme             from '../../../shared/theme'

describe('ExampleComponent', () => {
  it('component renders correctly', () => {

    const component = render(
      <ThemeProvider theme={theme}>
        <ExampleComponent />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
