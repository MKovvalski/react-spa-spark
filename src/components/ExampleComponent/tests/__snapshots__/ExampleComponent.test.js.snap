// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent component renders correctly 1`] = `
.emotion-14 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  font-size: 20px;
  margin-top: 15px;
}

.emotion-0 {
  font-size: 34px;
  color: #6699ff;
  margin-bottom: 15px;
  -webkit-transition: color .5s;
  transition: color .5s;
}

@media (max-width:74.99875em) {
  .emotion-0 {
    color: #00cc66;
  }
}

.emotion-10 {
  list-style-type: circle;
  font-size: 20px;
  margin-bottom: 15px;
  color: #000000;
}

.emotion-2 {
  margin-bottom: 5px;
  font-family: sans-serif;
}

.emotion-12 {
  background-color: #6699ff;
  padding: 15px;
  border: none;
  -webkit-transition: all .5s;
  transition: all .5s;
  cursor: pointer;
  border-radius: 10px;
  font-size: 16px;
}

.emotion-12:hover {
  background-color: #00cc66;
}

<div
  class="emotion-14 emotion-15"
>
  <h3
    class="emotion-0 emotion-1"
  >
    Aloha and welcome to React SPA spark!
  </h3>
  <ul
    class="emotion-10 emotion-11"
  >
    <li
      class="emotion-2 emotion-3"
    >
      emotion replaced sass
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      stylelint replaced sass-lint
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      removed code overhead from webpack configs
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      created a example file structure for emotion
    </li>
  </ul>
  <button
    class="emotion-12 emotion-13"
  >
    I am a common button
  </button>
</div>
`;
